### PR TITLE
[move-spec-test] benchmarks added

### PR DESCRIPTION
**Pull request by asmie**
### Description

Benchmarking code added.

Benchmarks are displayed on the `RUST_LOG` _info_ level.

### Test Plan
New unit tests for benchmark module added.

### CODE REVIEW

File: third_party/move/tools/move-spec-test/src/lib.rs [hunk 1]
```diff
@@ -55,11 +60,13 @@ pub fn run_spec_test(
 
     fs::create_dir_all(&outdir_original)?;
 
+    benchmarks.mutator.start();
     let outdir_mutant = if let Some(mutant_path) = &options.use_generated_mutants {
         mutant_path.clone()
     } else {
         run_mutator(options, config, &package_path, &outdir)?
     };
+    benchmarks.mutator.stop();
```
Comment (Rqnsom) [inline third_party/move/tools/move-spec-test/src/lib.rs]: Maybe run the benchmarks only around 
```rust
run_mutator(options, config, &package_path, &outdir)?
```
and print 0.00 for the `use_generated_mutants` option?

File: third_party/move/tools/move-spec-test/src/lib.rs [hunk 2]
```diff
@@ -105,7 +113,10 @@ pub fn run_spec_test(
             ));
         }
 
+        let mut proving_timer = Benchmark::new();
         let result = prove(config, &outdir_prove, &prover_conf, &mut error_writer);
+        proving_timer.stop();
+        benchmarks.prover_results.push(proving_timer);
```
Comment (Rqnsom) [inline third_party/move/tools/move-spec-test/src/lib.rs]: Optimization improvement possibility:

1. create all `proving_timer` benchmarks, so `report.get_mutants().len()` timers
2. `benchmarks.prover.start();` and run `for (elem, proving_timer_benchmark) in xxx`
3. proving_timer_benchmark.start()
4. prove elem
5. proving_timer_benchmark.stop()
6. benchmarks.prover.stop();
7. push all `proving_timer_benchmark` into `benchmarks.prover_results`

**Review by Rqnsom — APPROVED** [review]
LGTM with some suggestions (I don't think the optimization suggestion brings much value tbh)

File: third_party/move/tools/move-spec-test/src/lib.rs (hunk 1 above)
Comment (asmie) [reply third_party/move/tools/move-spec-test/src/lib.rs]: Sure.
Done.

File: third_party/move/tools/move-spec-test/src/lib.rs (hunk 2 above)
Comment (asmie) [reply third_party/move/tools/move-spec-test/src/lib.rs]: Nice remark!
Done :smile:

File: third_party/move/tools/move-spec-test/src/lib.rs [hunk 3]
```diff
@@ -120,6 +139,9 @@ pub fn run_spec_test(
         }
     }
 
+    benchmarks.prover.stop();
+    benchmarks.prover_results.extend(proving_benchmark);
```
Comment (Rqnsom) [inline third_party/move/tools/move-spec-test/src/lib.rs]: I suggest renaming it to `proving_benchmarks`
```suggestion
    benchmarks.prover_results = proving_benchmarks;
```

File: third_party/move/tools/move-spec-test/src/lib.rs [hunk 4]
```diff
@@ -58,7 +66,10 @@ pub fn run_spec_test(
     let outdir_mutant = if let Some(mutant_path) = &options.use_generated_mutants {
         mutant_path.clone()
     } else {
-        run_mutator(options, config, &package_path, &outdir)?
+        benchmarks.mutator.start();
+        let res = run_mutator(options, config, &package_path, &outdir)?;
+        benchmarks.mutator.stop();
+        res
```
Comment (Rqnsom) [inline third_party/move/tools/move-spec-test/src/lib.rs]:
```suggestion
        benchmarks.mutator.start();
        let outdir_mutant = run_mutator(options, config, &package_path, &outdir)?;
        benchmarks.mutator.stop();
        outdir_mutant
```

**Review by Rqnsom — APPROVED** [review]

File: third_party/move/tools/move-spec-test/src/lib.rs (hunk 3 above)
Comment (asmie) [reply third_party/move/tools/move-spec-test/src/lib.rs]: Done.

File: third_party/move/tools/move-spec-test/src/lib.rs (hunk 4 above)
Comment (asmie) [reply third_party/move/tools/move-spec-test/src/lib.rs]: Done.

**Pull request merged**